pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -281,3 +307,151 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
 		Providers: providers,
 	}
 }
+
+// getMutualTLSMode returns the MutualTLSMode enum corresponding peer MutualTLS settings.
+// Input cannot be nil.
+func getMutualTLSMode(mtls *v1beta1.PeerAuthentication_MutualTLS) model.MutualTLSMode {
+	switch mtls.Mode {
+	case v1beta1.PeerAuthentication_MutualTLS_DISABLE:
+		return model.MTLSDisable
+	case v1beta1.PeerAuthentication_MutualTLS_PERMISSIVE:
+		return model.MTLSPermissive
+	case v1beta1.PeerAuthentication_MutualTLS_STRICT:
+		return model.MTLSStrict
+	default:
+		return model.MTLSUnknown
+	}
+}
+
+// composePeerAuthentication returns the effective PeerAuthentication given the list of applicable
+// configs. This list should contains at most 1 mesh-level and 1 namespace-level configs.
+// Workload-level configs should not be in root namespace (this should be guaranteed by the caller,
+// though they will be safely ignored in this function). If the input config list is empty, returns
+// nil which can be used to indicate no applicable (beta) policy exist in order to trigger fallback
+// to alpha policy. This can be simplified once we deprecate alpha policy.
+// If there is at least one applicable config, returns should be not nil, and is a combined policy
+// based on following rules:
+// - It should have the setting from the most narrow scope (i.e workload-level is  preferred over
+// namespace-level, which is preferred over mesh-level).
+// - When there are more than one policy in the same scope (i.e workload-level), the one with
+// stronger security (STRICT > PERMISSIVE > DISABLE) is preferred.
+// - In a tie, the first in the list will be used (typically, the list is passed in sorted order by
+// timestamp, so it's more or less the latest).
+// - Port-level setting will be combined in similar manner: if there are more than 1 policy define
+// port-level mTLS for the same port, the stronger one is used.
+// - UNSET will be replaced with the setting from the parrent. I.e UNSET port-level config will be
+// replaced with config from workload-level, UNSET in workload-level config will be replaced with
+// one in namespace-level and so on.
+func composePeerAuthentication(rootNamespace string, configs []*model.Config) *v1beta1.PeerAuthentication {
+	var meshPolicy, namespacePolicy *v1beta1.PeerAuthentication
+
+	// Track number of workload level policies for early return.
+	numWorkloadLevelPolicies := 0
+
+	// First round to find mesh and namespace level policy.
+	for _, cfg := range configs {
+		spec := cfg.Spec.(*v1beta1.PeerAuthentication)
+		if cfg.Namespace == rootNamespace && spec.Selector == nil {
+			meshPolicy = spec
+		} else if spec.Selector == nil {
+			namespacePolicy = spec
+		} else if cfg.Namespace != rootNamespace {
+			numWorkloadLevelPolicies++
+		}
+	}
+
+	if meshPolicy == nil && namespacePolicy == nil && numWorkloadLevelPolicies == 0 {
+		return nil
```

Comment:
This should be defaulting to PERMISSIVE and not nil

Reply:
It will be treated as permissive when it's nil. The reason I use nil so that we know whether beta policy is available or not in order to fallback to alpha. It can be simplified later when we deprecate alpha completely.

Reply:
defaulting can be achieved via installation manifest including a permissive mesh wide policy. that's how we are doing today.